### PR TITLE
feat: [419250] Enable sending email from preview links

### PR DESCRIPTION
**Pull request by KiranVarma02**

### CODE REVIEW

File: src/server/plugins/engine/pageControllers/SummaryPageController.ts [hunk 1]
```diff
@@ -342,6 +343,10 @@ export function getPersonalisation(
     model
   )
 
+  const formName = isPreview
+    ? `TEST FORM SUBMISSION: ${model.name}`
+    : `Form received: ${model.name}`
```
Comment (alexluckett) [inline src/server/plugins/engine/pageControllers/SummaryPageController.ts:348]: This is not a form name any more but a subject line instead. Rename to `subject`?

File: src/server/plugins/engine/pageControllers/SummaryPageController.ts [hunk 2]
```diff
@@ -302,8 +298,12 @@ async function sendEmail(
 ) {
   request.logger.info(['submit', 'email'], 'Preparing email')
 
+  const { path } = request
+
+  const isPreview = path.toLowerCase().startsWith(PREVIEW_PATH_PREFIX)
```
Comment (alexluckett) [inline src/server/plugins/engine/pageControllers/SummaryPageController.ts:303]: We're duplicating this logic. Better to pull this out into a shared function so they don't drift.

Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/SummaryPageController.ts:303]: ahh ok, i didn't see its used in `plugin.ts`. I will refactor that

Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/SummaryPageController.ts]: done

File: src/server/plugins/engine/pageControllers/SummaryPageController.ts (hunk 1 above)
Comment (KiranVarma02) [reply src/server/plugins/engine/pageControllers/SummaryPageController.ts:348]: done

**Review by alexluckett — APPROVED** [review]

File: src/server/plugins/engine/helpers.ts [hunk 3]
```diff
@@ -77,3 +79,7 @@ export const filesize = (bytes: number) => {
 
   return Math.max(bytes, 0.1).toFixed(1) + byteUnits[i]
 }
+
+export const checkIfPreview = (path: string) => {
```
Comment (colinrotherham) [inline src/server/plugins/engine/helpers.ts:83]: Mind if we use a named function for stack traces and add some tests?

```suggestion
export function hasPreviewPath(path: string) {
```

File: src/server/plugins/engine/pageControllers/SummaryPageController.test.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+import { format } from 'date-fns'
+
+import { getPersonalisation } from './SummaryPageController.js'
+
+import {
+  FormModel,
+  SummaryViewModel
+} from '~/src/server/plugins/engine/models/index.js'
+
+describe('getPersonalisation', () => {
```
Comment (colinrotherham) [inline src/server/plugins/engine/pageControllers/SummaryPageController.test.ts:16]: Think the old description has been left in?

Or if not, let's move and rename this describe block

```suggestion
describe('SummaryPageController', () => {
  describe('getPersonalisation', () => {
```

File: src/server/plugins/engine/pageControllers/SummaryPageController.test.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import {
+  ComponentType,
+  type ComponentDef,
+  type FormDefinition
+} from '@defra/forms-model'
+import { type Request } from '@hapi/hapi'
+import { format } from 'date-fns'
+
+import { getPersonalisation } from './SummaryPageController.js'
```
Comment (colinrotherham) [inline src/server/plugins/engine/pageControllers/SummaryPageController.test.ts:9]: Another path for `~/src` please (sorry)

File: src/server/plugins/engine/pageControllers/SummaryPageController.ts (hunk 1 above)
Comment (colinrotherham) [reply src/server/plugins/engine/pageControllers/SummaryPageController.ts:348]: Just flagging this because it still says `formName`?

**Review by colinrotherham — CHANGES_REQUESTED** [review]

**Review by colinrotherham — APPROVED** [review]

**Pull request merged**